legion/engine/physics/data/colliders/collider_data.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <core/core.hpp>
+#include <physics/data/component_flags.hpp>
+#include <bitset>
+
+namespace legion::physics
+{
+    class ColliderData
+    {
+    public:
+
+        inline void setRegistered(bool registeredState) { m_isRegistered = registeredState; }
+        inline bool isRegistered() const { return m_isRegistered; }
+
+        inline const math::vec3& getOffset() const  { return m_positionOffset; }
+        inline const math::quat& getRotationOffset() const { return m_rotationOffset; }
+
+        inline collider_type getColliderType() const { return m_colliderType; }
+
+        inline bool isRegisteredOfType(collider_type colliderType) const { return m_isRegistered && colliderType == m_colliderType; }
+        
+    protected:
+
+        ColliderData(collider_type colliderType, const math::vec3& offset, const math::quat& rotation,
+            const std::bitset < physics_component_flag::pc_max>* modificationBitset);
```

Comment:
is the pointer passed to this function ever allowed to be null? if so then this is fine, if not, then you might want to ask for a reference, and then initialise the pointer in this struct with the address of the reference. However, since this struct doesn't have any other constructors anyways, it's also not default, move, or copy constructible. so then you might as well make `m_bitsetPtr` a `const std::bitset<physics_component_flag::pc_max>&` instead